### PR TITLE
Remove existing rename type to match file

### PR DESCRIPTION
**Pull request by jmarolf**
simplifying rename implementation to prepare for more changes

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs [hunk 1]
```diff
@@ -78,18 +83,131 @@ public async void OnWorkspaceChangedAsync(object sender, WorkspaceChangeEventArg
...
+            if (renamedSolution == null)
+                return;
+
+            await _threadingService.SwitchToUIThread();
+            bool renamedSolutionApplied = _roslynServices.ApplyChangesToSolution(project.Solution.Workspace, renamedSolution);
+
+            if (!renamedSolutionApplied)
+            {
+                string failureMessage = string.Format(CultureInfo.CurrentCulture, VSResources.RenameSymbolFailed, oldNameBase);
+                await _threadingService.SwitchToUIThread();
```
Comment (tmeschter) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs]: Is this redundant?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs [hunk 2]
```diff
@@ -78,18 +83,131 @@ public async void OnWorkspaceChangedAsync(object sender, WorkspaceChangeEventArg
...
+            }
+
+            return solutionToRename;
+        }
+
+        private static Document GetDocument(Project project, string filePath) =>
+            (from d in project.Documents where StringComparers.Paths.Equals(d.FilePath, filePath) select d).FirstOrDefault();
+
+        private static Task<SyntaxNode> GetRootNode(Document newDocument) =>
+            newDocument.GetSyntaxRootAsync();
```
Comment (tmeschter) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs]: This method seems unnecessary.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs [hunk 3]
```diff
@@ -78,18 +83,131 @@ public async void OnWorkspaceChangedAsync(object sender, WorkspaceChangeEventArg
...
+
+        private async Task<bool> CheckUserConfirmation(string oldFileName)
+        {
+            if (_userPromptedOnce)
+            {
+                return _userConfirmedRename;
+            }
+
+            await _threadingService.SwitchToUIThread();
+            bool userNeedPrompt = _environmentOptions.GetOption("Environment", "ProjectsAndSolution", "PromptForRenameSymbol", false);
```
Comment (tmeschter) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs]: Use a named parameter to clarify what `false` represents.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs (hunk 1 above)
Comment (jmarolf) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs]: No, tests will fail if this is not present

Comment (jmarolf) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs]: I can try and find a way to unify this logic though

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs (hunk 2 above)
Comment (jmarolf) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/Renamer.cs]: This was in the previous implementation.  I will remove/simplify this logic in a future PR

**Pull request merged**